### PR TITLE
Server: Firebase user auth

### PR DESCRIPTION
**Pull request by asclines**
## Major Updates
This Pull Request brings an update to the Server API, namely, the user-auth section.
This is part of an ongoing sub-project to utilize Firebase as a dev backend database while the real Backend is being built.

Methods that are implemented for Firebase and tested in this PR are:
- `login(email, password)`
- `logout()`

## Minor Updates
- Karma now shows logger output
- Update to `eslintrc.js` to ignore certain rules that are the result of Webpack.
- a6b52627a259ee78d54affd493165de47ebcb8cb Brings in `Firebase.js` that handles initializing the Firebase database. Integrated this into `App.js`
- Documentation is up to date with currently tested APIs.

## Final Notes
- The user-auth here is *not* implemented in the React App yet, just provided for the owners of those features to implement.

### CODE REVIEW

**Comment by asclines** [comment]
@NilsG-S I was going to leave the actual changes inside the UI part of the app up to you as you are more familiar with it. I can take on that role if you wish to take a more "advisory" role on this task.

**Comment by NilsG-S** [comment]
@asclines I can make the changes. It's just a matter of removing `firebase-config.js` and replacing `config.initializeApp(config)` with your `init` method, right?

**Comment by asclines** [comment]
@NilsG-S Sounds good, wanna just add that commit to this PR?

File: src/App.js [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
+// import logger from 'logger/logger.js';
+
+init();
 
-// Connects the app to firebase
-firebase.initializeApp(config);
 // store holds the redux store that allows app-wide state to be shared
 const store = createStore(cpceedApp);
-
+// logger.info('SERVER', init);
```
Comment (NilsG-S) [inline src/App.js]: Why leave the loggers there if they're commented?

Comment (asclines) [inline src/App.js]: Good catch. Will delete.

**Pull request merged**